app/scripts/nav.js

```diff
@@ -27,18 +29,33 @@ $(function() {
 
     lastScrollTop = currentScrollTop;
 
-    if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClass) ||
-        (!isOnTop(currentScrollTop) && nav.hasClass(onTopClass)))
-      toggleNavOnTop();
+    if (nav.hasClass(onTopClassIndex)) {
+      last_class = onTopClassIndex;
```

Comment:
The problem is that the Nav bar must behave differently if it is in the Index page (no logo on the left side) or in another page (logo appears on the left side). As such I believe I must have a variable outside the function to remember which page we are on. 
For example, in your code if I am in the Index page and I scroll down it would remove the --onTopIndex class. If I scroll up again it would toggle the --onTop class and the logo would appear on the Index page.

Also I agree that it makes sense to extract the second conditional to a function.